core/samples/net/tcp/main.cpp

```diff
@@ -0,0 +1,67 @@
+#include <cstring>
+#include <iostream>
+#include <thread>
+
+#include <cubos/core/log.hpp>
+/// [Import headers]
+#include <cubos/core/net/tcp_listener.hpp>
+#include <cubos/core/net/tcp_stream.hpp>
+/// [Import headers]
+#include <cubos/core/reflection/external/cstring.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+
+using cubos::core::net::Address;
+using cubos::core::net::TcpListener;
+using cubos::core::net::TcpStream;
+
+/// [Define constants]
+#define SERVER_ADDRESS Address::LocalHost
+#define SERVER_PORT 8080
+/// [Define constants]
+
+/// [Server function]
+void runServer()
+{
+    TcpListener listener;
+    listener.listen(SERVER_ADDRESS, SERVER_PORT, 1);
+
+    CUBOS_DEBUG("Server is listening");
+
+    TcpStream clientStream;
+    while (listener.accept(clientStream))
+    {
+        char buffer[1024];
+        if (std::size_t bytesRead = clientStream.read(buffer, sizeof(buffer)); bytesRead > 0)
```

Comment:
There's no guarantee that you will read the entire message at once. You should repeat read until `bytesRead = 0`.

Reply:
Im confused, again. Will `read` handle this? Or does the client need to?

Reply:
Assuming your discord message was after this, I'm gonna assume `TcpStream::read` has to do it.